### PR TITLE
Doc for metadata widget

### PR DESCRIPTION
**Pull request by remilejeune**

### CODE REVIEW

File: source/includes/_form_widgets.md [hunk 1]
```diff
@@ -37,20 +37,21 @@ Some of these API will contain a `widget` attribute that contains all you need t
 Attribute | Description
 --------- | -----------
 `name` <br> *string* | Identifier for the object
```
Comment (meganelacheny) [inline source/includes/_form_widgets.md]: "Identifier **of** the object"

File: source/includes/_form_widgets.md [hunk 2]
```diff
@@ -37,20 +37,21 @@ Some of these API will contain a `widget` attribute that contains all you need t
...
-`type` <br> *string* | Nature of the object. <br> Possible values are `text`, `multitext`, `choice`, `multichoice`, `date`, `datetime`, `html`, `boolean` and `number`
-`label` <br> *string* | Plain english name of the object
+`type` <br> *string* | Nature of the object. <br> Possible values are `text`, `multitext`, `choice`, `multichoice`, `date`, `datetime`, `html`, `boolean`, `int`, `number`, `geographicarea`
+`label` <br> *string* | Plain name of the object depending of the language
 `widget` <br> *[widget object](#the-widget-object)* | Characteristics of the expected rendered UI component for the object
 `choices` <br> *array of strings* | *`choice` and `multichoice` only* List of all accepted values by the object, any other value will be rejected.
```
Comment (meganelacheny) [inline source/includes/_form_widgets.md]: Could you add a line break after "*[...] only*"?
And the same goes for the other rows of this table!

File: source/includes/_form_widgets.md [hunk 3]
```diff
@@ -37,20 +37,21 @@ Some of these API will contain a `widget` attribute that contains all you need t
...
 
 ## The widget object
 
 ### Common attributes
 
 Attribute | Description
 --------- | -----------
-`type` <br> *string* | The widget's type <br> Accepted values are `textinput`, `select`, `datalist`, `multitextinput`, `multiselect`, `multidatalist`, `tags`, `dateinput`, `datetimeinput`, `toggle` and `number`
+`type` <br> *string* | The widget's type <br> Accepted values are `textinput`, `select`, `datalist`, `multitextinput`, `multiselect`, `multidatalist`, `tags`, `dateinput`, `datetimeinput`, `toggle`, `number` and `geographicarea`
 `help_text` <br> *string* | Help text describing the constraints and uses of this form object
```
Comment (meganelacheny) [inline source/includes/_form_widgets.md]: Writing in a description that "help_text" is a help text a little bit too obvious :)
"Informational text describing the constraints and uses of the form object", for example, could be a better description.

File: source/includes/_form_widgets.md [hunk 4]
```diff
@@ -174,12 +175,182 @@ Attribute | Description
 `orderable` <br> *boolean* | Flag indicated whether the visual component should allow values to be reordered or not
```
Comment (meganelacheny) [inline source/includes/_form_widgets.md]: "Flag indicat**ing**"

File: source/includes/_form_widgets.md [hunk 5]
```diff
@@ -174,12 +175,182 @@ Attribute | Description
...
 
 #### The `multiselect` widget
+
+The `multiselect` widget is to be rendered as multiple select elements, with the possibility to remove and add more inputs. If the `orderable` option is set, then the inputs must be reorderable one relative to the other.
+Options of this element must match the `choices` defined in the parent [form object](#the-form-object).
+
+Attribute | Description
+--------- | -----------
+`orderable` <br> *boolean* | Flag indicated whether the visual component should allow values to be reordered or not
```
Comment (meganelacheny) [inline source/includes/_form_widgets.md]: "Flag indicat**ing**"

File: source/includes/_form_widgets.md [hunk 6]
```diff
@@ -174,12 +175,182 @@ Attribute | Description
...
+
 #### The `multidatalist` widget
+
+The `multidatalist` widget is to be rendered as multiple text inputs with an autocomplete feature, with the possibility to remove and add more inputs. If the `orderable` option is set, then the inputs must be reorderable one relative to the other.
+
+Attribute | Description
+--------- | -----------
+`orderable` <br> *boolean* | Flag indicated whether the visual component should allow values to be reordered or not
```
Comment (meganelacheny) [inline source/includes/_form_widgets.md]: "Flag indicat**ing**"

File: source/includes/_form_widgets.md [hunk 7]
```diff
@@ -174,12 +175,182 @@ Attribute | Description
...
+> Example HTML widget
+
+```html
+<label for="last_modified">Last Modified</label>
+<input id="last_modified" type="datetime-local">
+```
+
 #### The `toggle` widget
+
+The `toggle` widget is to be rendered as a ods-toggle input.
```
Comment (meganelacheny) [inline source/includes/_form_widgets.md]: This introduction (as well as most of the others) could be improved. We need to make sure that everyone, no matter their "technical level", at least understands the introductions and descriptions.

File: source/includes/_form_widgets.md [hunk 8]
```diff
@@ -174,12 +175,182 @@ Attribute | Description
...
+> Example object
+
+```json
+{
+    "name": "language",
+    "type": "text",
+    "label": "Language",
+    "choices": ["fr", "en"],
+    "widget": {
+        "type": "radio"
```
Comment (meganelacheny) [inline source/includes/_form_widgets.md]: I think that's the example for the radio widget and not for the number widget.

**Review by meganelacheny — CHANGES_REQUESTED** [review]

**Comment by jmorel** [comment]
[ch9710]

**Comment by jmorel** [comment]
@nicoconstanzo I'm done with the doc, I'd like you to check the widgets examples and tell me if they are understandable.

**Comment by nicoconstanzo** [comment]
@jmorel 
- Since you're suggesting the help_text as a parameter, why not use it in the first example to show the output?

- The `<select>` options are a bit criptic (I know they're licenses but still, something in the lines of Free, Personal, Professional?). In the object, I think it would be more clear putting the "widget" key before the "choices".
screen shot 2017-07-20 at 17 48 24

 Wording: 
- *The `textinput` widget is to be rendered as a simple text input.*
 The **simple** word conveys that there's a complex option. Just saying text input should be enough. 
"is to be rendered" sounds contrived => will render, renders, outputs.
- Instead of **Example HTML widget** as the example title, why not use something like **Object HTML output**?

**Comment by jmorel** [comment]
> Since you're suggesting the help_text as a parameter, why not use it in the first example to show the output?

You're right, I should use it. But in all examples.
edit: fun fact, we almost never have help text set in the platform. In order to have real life examples, I'll stick to what the API actually outputs.

> The `<select>` options are a bit criptic (I know they're licenses but still, something in the lines of Free, Personal, Professional?). In the object, I think it would be more clear putting the "widget" key before the "choices".

The only order that is relevant is the one we return in the platform. I should check out which it is. 
About the select options, you're right. I should either go for an example straight from the platform or a more symbolic one.

> Wording:

I'll update it. However I'll keep the "Example" as it is to stay in line with the rest of the wording of the right column (definition / example object / example request / example response)

**Review by meganelacheny — APPROVED** [review]

**Pull request merged**